src/app/app.component.ts

```diff
@@ -57,7 +53,14 @@ export class AppComponent implements OnInit, OnDestroy {
 
 	readonly chartData$: Observable<ChartData>;
 
-	readonly packageNames: ArrayObservable<string> = new ArrayObservable(this.getCachedPackageNames('package-names'));
+	/* loaded package names  */
+	readonly packageNames = new FormControl<string[]>(this.getCachedPackageNames('package-names'), {
```

Comment:
It looks like the mat-error isn't working for the auto complete component. We'll have to pass the packageNames to it somehow to get the error handler to work properly or find a way to pass the validation from AppComponent.

Because of this, I don't think we should have refactored packageNames from `ArrayObservable` to `FormControl`. Relying on FormControl.valueChanges has its cons and it'll make things harder to implement overall.